src/components/map/Map.tsx

```diff
@@ -105,22 +60,60 @@ const Map = ({
       );
   }
 
-  const flyToFeature = (feature: Feature) => {
+  const onPress = async (feature: Feature) => {
+    if (shouldExploreTripFromStops) {
+      await selectStopPlace(feature);
+    } else if (shouldSelectLocation) {
+      selectPoint(feature);
+    }
+  };
+
+  const selectPoint = (feature: Feature) => {
     if (feature && feature.geometry.type === 'Point') {
+      setSelectedCoordinates({
+        longitude: feature.geometry.coordinates[0],
+        latitude: feature.geometry.coordinates[1],
+      });
       mapCameraRef.current?.flyTo(feature.geometry.coordinates, 300);
     }
   };
 
-  const styles = useMapStyles();
-  const controlStyles = useControlPositionsStyle();
+  async function selectStopPlace(feature: any) {
+    const {screenPointX, screenPointY} = feature.properties;
+    const renderedFeaturesResult =
+      await mapViewRef?.current?.queryRenderedFeaturesAtPoint(
+        [screenPointX, screenPointY],
+        ['==', ['geometry-type'], 'Point'],
+      );
+
+    renderedFeaturesResult?.features.map(async (featureAtPoint) => {
+      if (
+        featureAtPoint.geometry.type === 'Point' &&
+        featureAtPoint?.properties?.entityType === 'StopPlace'
+      ) {
+        setSelectedCoordinates({
+          longitude: featureAtPoint.geometry.coordinates[0],
+          latitude: featureAtPoint.geometry.coordinates[1],
+        });
+        mapCameraRef.current?.fitBounds(
+          [coordinates.longitude, coordinates.latitude],
+          featureAtPoint.geometry.coordinates,
+          [100, 100],
+          1000,
+        );
+      }
+    });
+  }
```

Comment:
yep, i intended to remove forEach but forgot 😅  
There is an if-block to take care of the fact that only StopPlaces are considered, but I can move it to find so that we stop when the StopPlace is found.